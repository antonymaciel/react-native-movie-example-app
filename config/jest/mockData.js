export const movie = {
    id: 0,
    year: '2019',
    imageUrl: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    title: 'Spiderman',
    vote_average: '9.1',
    urlbackdropImage: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg', 
    generes: 'Aventure, Drama, Terror',  
    vote_count: '1200',
    overview: 'Amazing movie',
    backdrop_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    genres: ['Adventure', 'Terror', 'Drama'],
    release_date: '2016-5-14'
}

export const review = {
    id: 0,
    author: 'Antony Maciel',
    url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
    content: 'Great Movie!, always has amazing content \n I love this movie'
}

export const movies = [
    {
        id: 1,
        poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'XMan',
        vote_average: 7.8,
        release_date: '2016-5-14'
    },
    {     
        id: 2,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Superman',
        vote_average: 0.9,
        release_date: '2015-5-9'
    },
    {
        id: 3,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Spiderman',
        vote_average: 8.0,
        release_date: '2005-5-27'
    }
]


export const reviews = [
    {
        id: 1,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!!!'
    },
    {
        id: 2,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!, always has amazing content \n I love this movie'
    },
    {
        id: 3,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: '&^%$#@!'
    }
]

