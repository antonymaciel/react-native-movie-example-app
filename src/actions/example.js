

import Api from "../api/apiMovies"

const action = () => {
    Api.getMovieReviews('429617')
    .then(response => 
        console.log('RESPONSE', response)
    )
    .catch(err => {
    console.log(err)
    })
    return 'hola'
  }
 