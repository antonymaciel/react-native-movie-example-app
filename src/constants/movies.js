export const MOVIES_SUCCESS = 'MOVIES_SUCCESS';
export const MOVIES_FAILD = 'MOVIES_FAILD';
export const MOVIES_LOADING = 'MOVIES_LOADING';
export const MOVIES_CLEAN = 'MOVIES_CLEAN';