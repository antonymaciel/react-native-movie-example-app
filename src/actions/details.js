import Api from "../api/apiMovies"
import { DETAILS_SUCCESS, DETAILS_FAILD, DETAILS_LOADING, DETAILS_FINISH } from '../constants/details'

const movieDetailsSuccess = (movieDetails) => ({
    type: DETAILS_SUCCESS,
    movieDetails
});

const movieDetailsFaild = () => ({
    type: DETAILS_FAILD, 
});

const movieDetailsLoading = () => ({
    type: DETAILS_LOADING,
});

export const movieDetailsFinish = () => ({
    type: DETAILS_FINISH,
});

export const getMovieDetails = (idMovie) => (dispatch) => {
    dispatch(movieDetailsLoading());
    Api.getMovieDetails(idMovie)
    .then((json) => {
        dispatch(movieDetailsSuccess(json));
    })
    .catch(err => {
        console.log(err);
        dispatch(movieDetailsFaild());
    })
  };