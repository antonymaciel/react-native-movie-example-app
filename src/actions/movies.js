import Api from "../api/apiMovies"
import { MOVIES_SUCCESS, MOVIES_FAILD, MOVIES_LOADING } from '../constants/movies'

export const moviesSuccess = (movies) => ({
    type: MOVIES_SUCCESS,
    movies 
});

export const moviesFaild = () => ({
    type: MOVIES_FAILD, 
});

export const moviesLoading = () => ({
    type: MOVIES_LOADING,
});

export const getMovies = (page) => (dispatch) => {
    dispatch(moviesLoading());
    Api.getMovies(page)
    .then((json) => {
        dispatch(moviesSuccess(json));
    })
    .catch(err => {
        console.log(err);
        dispatch(moviesFaild());
    })
  };
 