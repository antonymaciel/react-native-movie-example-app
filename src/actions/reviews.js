import Api from "../api/apiMovies"
import { REVIEWS_SUCCESS, REVIEWS_FAILD, REVIEWS_LOADING, REVIEWS_FINISH } from '../constants/reviews'

export const movieReviewsSuccess = (reviews) => ({
    type: REVIEWS_SUCCESS,
    reviews
});

export const movieReviewsFaild = () => ({
    type: REVIEWS_FAILD, 
});

export const movieReviewsLoading = () => ({
    type: REVIEWS_LOADING,
});

export const movieReviewsFinish = () => ({
    type: REVIEWS_FINISH,
});

export const getMovieReviews = (idMovie, page) => (dispatch) => {
    dispatch(movieReviewsLoading());
    Api.getMovieReviews(idMovie, page)
    .then((json) => {
        dispatch(movieReviewsSuccess(json));
    })  
    .catch(err => {
        console.log(err);
        dispatch(movieReviewsFaild());
    })
  };

