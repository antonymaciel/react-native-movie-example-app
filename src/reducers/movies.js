import { MOVIES_SUCCESS, MOVIES_FAILD, MOVIES_LOADING } from '../constants/movies'

const initialState = {
    movies: null,
    loading: false,
    error: false
}


const movies = (state = initialState, action) => {
    switch(action.type){
        case MOVIES_SUCCESS:
            console.log(action);
            const movies = action.movies.results;
            return {
                movies: state.movies ? [ ...state.movies, ...movies] : movies ,
                loading: false, 
                error: false
            };
        case MOVIES_LOADING:
            console.log(action);
            return { 
                ...state, 
                loading: true,
                error: false
            };
        case MOVIES_FAILD:
            console.log(action);
            return { 
                ...state, 
                loading: false, 
                error: true
            };
        default:
                return state;
    } 
}

export default movies;