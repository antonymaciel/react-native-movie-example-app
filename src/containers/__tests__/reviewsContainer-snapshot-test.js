import React from 'react';
import renderer from 'react-test-renderer';
import mockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { reviews } from '../../../config/jest/mockData';
import ReviewsContainer from '../reviewsContainer';
import * as actions from '../../actions/movies';
jest.mock('../../screens/reviews', () => 'Reviews');


it('renders reviews container using Snapshots', () => {
    const _getComponent = (props) => {
        const state = {movieReviews: props};
        let store = mockStore(state);
        store.dispatch = jest.fn();
        actions.getMovieReviews = jest.fn();
        actions.movieReviewsFinish = jest.fn();
        return (
            renderer.create(
                <Provider store={store} >
                    <ReviewsContainer 
                        navigation={{getParam: jest.fn(), navigate: jest.fn()}}
                    />
                </Provider>
            )
        )
    };

  let component = _getComponent({reviews: reviews, totalReviews: reviews.length ,loading: false});
  expect(component).toMatchSnapshot();
  
  
  component = _getComponent({reviews: null, totalReviews: 0, loading: true});
  expect(component).toMatchSnapshot();

  component = _getComponent({reviews: null, totalReviews: 0, loading: false});
  expect(component).toMatchSnapshot();
})