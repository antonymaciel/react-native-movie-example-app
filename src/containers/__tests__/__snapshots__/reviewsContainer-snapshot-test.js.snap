// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders reviews container using Snapshots 1`] = `
<Reviews
  onNewPage={[Function]}
  posterUrl="https://image.tmdb.org/t/p/w185undefined"
  results={
    Array [
      Object {
        "author": "Antony Maciel",
        "content": "Great Movie!!!",
        "id": 1,
        "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
      },
      Object {
        "author": "Antony Maciel",
        "content": "Great Movie!, always has amazing content 
 I love this movie",
        "id": 2,
        "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
      },
      Object {
        "author": "Antony Maciel",
        "content": "&^%$#@!",
        "id": 3,
        "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
      },
    ]
  }
  total_results={3}
/>
`;

exports[`renders reviews container using Snapshots 2`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#db4f48"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`renders reviews container using Snapshots 3`] = `
<Text>
  Error...
</Text>
`;
